src/components/ResourceBrowser/ResourceList/ResourceFilterOptions.tsx

```diff
@@ -86,14 +90,17 @@ const ResourceFilterOptions = ({
         }
     }, [registerShortcut, isOpen])
 
-    const handleFilterKeyPress = (e: React.KeyboardEvent): void => {
+    const handleFilterKeyUp = (e: KeyboardEvent): void => {
         if (e.key === 'Escape' || e.key === 'Esc') {
             searchInputRef.current?.blur()
         }
     }
 
-    const handleOnChangeSearchText: React.FormEventHandler<HTMLInputElement> = (event): void => {
-        setSearchText(event.currentTarget.value)
+    const handleOnChangeSearchText: ComponentProps<typeof SearchBar>['handleSearchChange'] = (text) => {
+        setSearchText(text)
+        if (!text) {
```

Comment:
Verify for autoFocus if there is any empty state from which we clear search

Reply:
There is an empty state that clears the search filter from outside the search bar. But it would be hard to do so leaving for now.